tools/PI/DevHome.PI/Helpers/WatsonHelper.cs

```diff
@@ -142,37 +116,21 @@ public List<WatsonReport> GetWatsonReports()
                 var moduleName = entry.ReplacementStrings[3];
                 var executable = entry.ReplacementStrings[0];
                 var eventGuid = entry.ReplacementStrings[12];
-                var report = new WatsonReport(timeGenerated, moduleName, executable, eventGuid);
-                reports.Add(entry.ReplacementStrings[12], report);
+
+                var watsonViewModel = Application.Current.GetService<WatsonPageViewModel>();
+                watsonViewModel.AddNewEntry(timeGenerated, moduleName, executable, eventGuid);
             }
             else if (entry.InstanceId == 1001
                 && entry.Source.Equals("Windows Error Reporting", StringComparison.OrdinalIgnoreCase))
             {
-                // See if we've already put this into our Dictionary.
-                if (reports.TryGetValue(entry.ReplacementStrings[19], out WatsonReport? report))
-                {
-                    report.WatsonLog = entry.Message;
-
-                    try
-                    {
-                        // List files available in the archive.
-                        if (Directory.Exists(entry.ReplacementStrings[16]))
-                        {
-                            var files = Directory.EnumerateFiles(entry.ReplacementStrings[16]);
-                            foreach (var file in files)
-                            {
-                                report.WatsonReportFile = File.ReadAllText(file);
-                            }
-                        }
-                    }
-                    catch
-                    {
-                    }
-                }
+                var eventGuid = entry.ReplacementStrings[19];
+                var watsonLog = entry.Message;
+                var directoryPath = entry.ReplacementStrings[16];
+
+                var watsonViewModel = Application.Current.GetService<WatsonPageViewModel>();

```

Comment:
So many `getService` calls.  Is it possible to move all these calls to the constructor?  Then the the services are retrieved only once.